docs/example_navigation_ros_codegen.md

```diff
@@ -0,0 +1,527 @@
+    .with_verbosity_level(1)
+builder.build()
+
+
+if doSim:
+    # Use TCP server
+    # ------------------------------------
+    mng = og.tcp.OptimizerTcpManager('optimization_engine/mpc_controller')
+    mng.start()
+
+    current_x_0 = [-1, -1, 0]
+    current_x_ref = [2, 2, 0]
+    current_u = [0, 0]
+    cur_opt_var = [0] * (N * NU)  # current optimization variables
+    cur_opt_par = current_x_0 + current_x_ref  # current optimization parameters
+    average_solvetime = 0.0
+
+    x_sim = np.zeros((NSim + 1, NX))
+    x_sim[0:N + 1] = current_x_0
+
+    for k in range(0, NSim):
+
+        solver_status = mng.call(cur_opt_par)
+        cur_opt_var = solver_status['solution']
+        solvetime = solver_status['solve_time_ms']
+        current_u = cur_opt_var[0:NU]
+        current_x_0 = dynamics_dt(current_x_0, current_u).toarray().flatten()
+        x_sim[k + 1] = current_x_0
+
+        print(("Step %d/%d took %f [ms]") % (k, NSim, solvetime))
+        average_solvetime += solvetime
+
+        cur_opt_par = list(current_x_0) + current_x_ref
+
+    mng.kill()
+
+    print(("Average solvetime %.4f [ms]") % (average_solvetime/NSim))
+
+    plt.figure()
+    plt.subplot(3,1,1)
+    plt.plot(x_sim[:, 0])
+    plt.subplot(3, 1, 2)
+    plt.plot(x_sim[:, 1])
+    plt.subplot(3, 1, 3)
+    plt.plot(x_sim[:, 0], x_sim[:, 1])
+    plt.show()
+```
+
+The above program will generate a parametric optimizer at `optimization_engine/mpc_controller`. The ROS package will be in `optimization_engine/mpc_controller/open_nmpc_controller`.
+
+## Create a ROS workspace for our new package
+
+We must create a ROS catkin workspace and place our package inside the `src` folder. We will do this by creating a symbolic link. After that, we can build and test our package.
+
+```bash
+source /opt/ros/melodic/setup.bash
+cd ~/open_ros_codegen/
+mkdir -p catkin_ws/src
+cd catkin_ws/src
+ln -s ../../nmpc_open/optimization_engine/mpc_controller/open_nmpc_controller .
+cd ~/open_ros_codegen/catkin_ws
+catkin build
+source devel/setup.bash
+roslaunch open_nmpc_controller open_optimizer.launch
+```
+
+Right now we have a running ROS node with the defualt configuration. Now we must edit it a little bit to drive our vehicle.
+
+## Download and run Husky packages
+
+```bash
+cd ~/open_ros_codegen/catkin_ws/src
+git clone https://github.com/husky/husky.git
+catkin build
+```
+Now we can run the Husky simulation as follows
+
+``roslaunch husky_gazebo husky_empty_world.launch``
+
+## Edit the auto-generated node
+
+Our node should know where the vehicle is in order to obtain the correct control commands, so we must listen to an Odometry topic that is being published to `/odometry/filtered`
+
+Given our current position and orientation, we will obtain the NMPC solution and publish it as a Twist message to the `/husky_velocity_controller/cmd_vel` topic.
+
+In order to do that, we must modify the auto generated `open_optimizer.cpp` file as follows:
+
```

Comment:
Say a few words about this: what are the associated topics? In what ways do you need to modify the auto-generated file?